apps/native/app/src/stores/auth-store.ts

```diff
@@ -62,54 +65,73 @@ export const authStore = create<AuthStore>((set, get) => ({
   cognitoDismissCount: 0,
   cognitoAuthUrl: undefined,
   cookies: '',
-  async fetchUserInfo(_refresh = false) {
+  async fetchUserInfo(skipRefresh = false) {
     const appAuthConfig = getAppAuthConfig()
     // Detect expired token
     const expiresAt = get().authorizeResult?.accessTokenExpirationDate ?? 0
+
     if (new Date(expiresAt) < new Date()) {
       await get().refresh()
     }
-    return fetch(
+
+    const res = await fetch(
       `${appAuthConfig.issuer.replace(/\/$/, '')}/connect/userinfo`,
       {
         headers: {
           Authorization: `Bearer ${get().authorizeResult?.accessToken}`,
         },
       },
-    ).then(async (res) => {
-      if (res.status === 401) {
-        // Attempt to refresh the access token
-        if (!_refresh && (await get().refresh())) {
-          // Retry the userInfo call
-          return get().fetchUserInfo(true)
-        }
-        throw new Error('Unauthorized')
-      } else if (res.status === 200) {
-        const userInfo = await res.json()
-        set({ userInfo })
-        return userInfo
+    )
+
+    if (res.status === 401) {
+      // Attempt to refresh the access token
+      const successfulRefresh = await get().refresh()
+
+      if (!skipRefresh && successfulRefresh) {
```

Comment:
The difference from before has to do with [short circuit behaviour](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND#short-circuit_evaluation) in `&&`.

Before it only called refresh if `_refresh === false`. Now it calls and waits for a refresh even though `skipRefresh === false` and the actual refresh result is ignored.

Reply:
Yeah good point! I will update